components/ui/checkbox.tsx

```diff
@@ -12,13 +16,40 @@ interface CheckboxProps {
   iconSize?: number;
 }
 
-const AnimatedCheck = Animated.createAnimatedComponent(Check);
+interface AnimatedCheckProps {
+  value: boolean;
+  size: number;
+  className: string;
+  opacity: number;
+}
+
+const AnimatedCheck = Animated.createAnimatedComponent(
+  React.forwardRef(
+    (
+      { value, size, className, opacity, ...props }: AnimatedCheckProps,
+      ref: React.ForwardedRef<any>
```

Comment:
I try to avoid `any`s as much as possible. You can look at other `forwardRef` examples for typing

Reply:
I've removed these kinds of typings, so I think that will resolve this request